src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -224,30 +222,14 @@ private function autowireMethod(\ReflectionFunctionAbstract $reflectionMethod, a
     /**
      * @return TypedReference|null A reference to the service matching the given type, if any
      */
-    private function getAutowiredReference(TypedReference $reference, $deprecationMessage)
+    private function getAutowiredReference(TypedReference $reference)
     {
         $this->lastFailure = null;
         $type = $reference->getType();
 
         if ($type !== (string) $reference || ($this->container->has($type) && !$this->container->findDefinition($type)->isAbstract())) {
             return $reference;
         }
-
-        if (!$reference->canBeAutoregistered()) {
```

Comment:
is `canBeAutoregistered` still used anywhere ? AFAICT, this PR removes the only usages of the method. We should probably deprecate it.

Reply:
> We should probably deprecate it.

Yes, we can now in 4.1